db/models/models.js

```diff
@@ -10,6 +10,39 @@ const fetchTopics = () => {
     })
 }
 
+const fetchArticles = () => {
+    const sqlString = 
+    `
+        SELECT articles.*, COUNT(comments.comment_id) as comment_count
+        FROM articles
+        LEFT JOIN comments
+        ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id
+        ORDER BY created_at DESC;
+    `
+    return db.query(sqlString)
+    .then(({rows}) => rows)
+}
+
+const fetchCommentsByArticleId = (article_id) => {
+    const sqlString = 
+    `
+        SELECT * FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC;
+    `
+    return db.query(sqlString, [article_id])
+    .then(({rowCount, rows}) => {
+        if (rowCount === 0) {
+            return Promise.reject({status: 404, message: 'Not Found'})
```

Comment:
There will be occasions where there will be articles with no comments. This will return a rowCount of 0. What we need to do if this is the case is check that the article exists, if it does we can return a 200 and an empty array. If it doesn't exist then a 404 - Not Found is fine